pgmpy/estimators/TreeSearch.py

```diff
@@ -100,87 +96,134 @@ def estimate(self):
+            return TreeSearch.chow_liu(
+                self.data, self.root_node, edge_weights, self.n_jobs, show_progress
+            )
 
-        data = self.data
-        n_jobs = self.n_jobs
-        root_node = self.root_node
-        return_type = self.return_type
-        class_node = self.class_node
+        # Step 3: If estimator_type = "tan":
+        elif estimator_type == "tan":
 
-        if return_type == "chow-liu":
-            # construct tree with chow-liu algorithm
-            return DAG(self.chow_liu(data, root_node, n_jobs))
+            # Step 3.1: Checks for class_node and root_node != class_node
+            if class_node is None:
+                raise ValueError("class_node must be specified for estimator_type=tiu")
 
-        else:
-            if class_node not in data.columns:
-                raise ValueError("Class node %s not found in data columns" % class_node)
+            elif class_node not in self.data.columns:
+                raise ValueError(f"Class node: {class_node} not found in data columns")
 
-            if root_node == class_node:
+            elif self.root_node == class_node:
                 raise ValueError(
-                    "Root node %s and class node %s are identical"
-                    % (root_node, class_node)
+                    f"Root node: {self.root_node} and class node: {class_node} are identical"
                 )
 
-            # construct tree with chow-liu algorithm
-            df_features = data.loc[:, data.columns != class_node]
-            D = self.chow_liu(df_features, root_node, n_jobs)
+            # Step 3.2:: Construct chow-liu on {data.columns - class_node}
+            df_features = self.data.loc[:, self.data.columns != class_node]
+            D = TreeSearch.chow_liu(
+                df_features, self.root_node, edge_weights, self.n_jobs, show_progress
+            )
+
+            # Step 3.3: Add edges from class_node to all other nodes.
+            D.add_edges_from([(class_node, node) for node in df_features.columns])
+
+            return D
+
+    @staticmethod
+    def chow_liu(
+        data, root_node, edge_weights="mutual_info", n_jobs=-1, show_progress=True
+    ):
+        """
+        Chow-Liu algorithm for estimating tree structure from given data. Refer to 
+        pgmpy.estimators.TreeSearch for more details.
+
+        Parameters
+        ----------
+        data: pandas.DataFrame object
+            dataframe object where each column represents one variable.
+
+        root_node: str, int, or any hashable python object.
+            The root node of the tree structure.
+
 
-            # add edge from class node to other nodes
-            for node in df_features.columns:
-                D.add_edge(class_node, node)
+        n_jobs: int (default: -1)
+            Number of jobs to run in parallel. `-1` means use all processors.
 
-            return DAG(D)
+        edge_weights: str or function (default: mutual info)
```

Comment:
I'm not sure what the convention is in `pgmpy` but I wonder, for clarity, should `edge_weights` be renamed to `edge_weights_fn` or something similar to hint that a function is expected.

Reply:
Yes, have changed it to `edge_weights_fn` now.